superset-frontend/src/explore/components/SaveModal.jsx

```diff
@@ -109,19 +110,31 @@ class SaveModal extends React.Component {
     sliceParams.save_to_dashboard_id = this.state.saveToDashboardId;
     sliceParams.new_dashboard_name = this.state.newDashboardName;
 
-    this.props.actions
-      .saveSlice(this.props.form_data, sliceParams)
-      .then(({ data }) => {
-        if (data.dashboard_id === null) {
-          sessionStorage.removeItem(SK_DASHBOARD_ID);
-        } else {
-          sessionStorage.setItem(SK_DASHBOARD_ID, data.dashboard_id);
-        }
-        // Go to new slice url or dashboard url
-        const url = gotodash ? data.dashboard_url : data.slice.slice_url;
-        window.location.assign(url);
+    if (slice.slice_updated) {
+      const payload = {
+        description: slice.description || null,
+        cache_timeout: slice.cache_timeout || null,
+        owners: slice.owners || null,
+      };
+      await SupersetClient.put({
```

Comment:
why have to add extra API call to save this change? if you remove this extra call, the change still be saved, right?

Reply:
No, that second call doesn't update data such as description and cache timeout. Essentially, we send the same number of requests as before - we just delayed sending a PUT request until the main "save" button is clicked.

Reply:
so these 2 modals hold different data and we have to use 2 different API save each of them? 
then this flow seems very intuitive: save something in one Modal, save it, Modal closed. 
Then user have to click SAVE again and open another Modal, seeing some not very relative data, and save them again.
If someone don't look at the source code and APIs, how do they understand this is right flow to save chart metadata?

Reply:
This flow was requested by Junlin: `general approach should be to save to the local state on Ok on modal and wait until the actual [more deliberate] Save action by the user to post to the db`
@junlincc WDYT?

Reply:
Personally I think an "Ok" button that affects the redux store is the way to go, and a single "Save" api call later on that saves the whole thing. Similarly, using the editable click-on-title should only affect the redux store.

Do we still need the "Edit Chart Properties" from any other contexts like the Chart list view? I don't think so. I'd just rip that out and assume that people have to go to the explore view if they want to edit the properties of a chart.

Reply:
> Do we still need the "Edit Chart Properties" from any other contexts like the Chart list view? I don't think so. I'd just rip that out and assume that people have to go to the explore view if they want to edit the properties of a chart.

A similar sentiment happened on https://github.com/apache/incubator-superset/pull/11667 - I'll change that one to remove (or just comment out) the Edit Chart Properties links in the list/card views.

Reply:
I see. In that case, I'm going to change the status of this PR to draft and wait for merge of https://github.com/apache/incubator-superset/pull/11667. Then I'll remove the conditional requests in Edit properties modal and rename "Save" to "Ok" in the same modal.
However, that doesn't solve the concerns of @graceguo-supercat regarding the flow of saving edited data through SaveModal. Or do we have a consensus in that matter?

Reply:
To make my concern short, in order to save any changes in Chart Properties modal, do we really need to 2nd API call?

Reply:
I think we do. POST request to `/superset/explore/?form_data...` doesn't save parameters that we can edit in PropertiesModal - cache_timeout and description. I tried to append these 2 to that request and it didn't work. Maybe a change on backend that would allow saving description and cache_timeout along with other chart data is in order?

Reply:
To **only** save changes in the Properties Modal, do you need to call `/superset/explore/?form_data`?

Reply:
Ah, I see what you mean. I'll try to optimize this flow.